test/Streamly/Test/Internal/Data/Fold.hs

```diff
@@ -13,15 +14,21 @@ maxStreamLen :: Int
 maxStreamLen = 1000
 
 testRollingHashFirstN :: Property
-testRollingHashFirstN = 
+testRollingHashFirstN =
     forAll (choose (0, maxStreamLen)) $ \len ->
         forAll (choose (0, len)) $ \n ->
             forAll (vectorOf len (arbitrary :: Gen Int)) $ \vec -> monadicIO $ do
                 a <- run $ S.fold rollingHash $ S.take n $ S.fromList vec
                 b <- run $ S.fold (rollingHashFirstN n) $ S.fromList vec
                 assert $ a == b
 
+testHead :: [Int] -> Expectation
+testHead ls = S.fold FL.head (S.fromList (ls :: [Int])) `shouldReturn` headl ls
+            where headl [] = Nothing
+                  headl (x:_) = Just x
+
 main :: IO ()
 main = hspec $
-    describe "Rolling Hash Folds" $
+    describe "Rolling Hash Folds" $ do
```

Comment:
Change the description? Both aren't rolling hash folds.

Reply:
Okay. I will.